src/main.rs

```diff
@@ -608,11 +720,20 @@ fn get_setup(args: &[String]) -> Setup {
     let initial_volume = matches
         .opt_str(INITIAL_VOLUME)
         .map(|initial_volume| {
-            let volume = initial_volume.parse::<u16>().unwrap();
+            let volume = initial_volume.parse::<u16>().unwrap_or_else(|_| {
+                error!("Invalid --{}: {}", INITIAL_VOLUME, initial_volume);
+                println!("Valid --{} values: 0-100", INITIAL_VOLUME);
+                println!("Default: 50");
```

Comment:
Or the current value when the Alsa mixer is used.

Reply:
This will get you the actual used backend's name. It would be trivial to then basically say: "Well you're not using the alsa backend you can't use any alsa specific functionality."

```rust
pub fn find_name(name: Option<String>) -> &'static str {
    if let Some(name) = name {
        BACKENDS
            .iter()
            .find(|backend| name == backend.0)
            .map(|backend| backend.0).unwrap_or_default()
    } else {
        BACKENDS.first().map(|backend| backend.0).unwrap_or_default()
    }
}
```

Reply:
Or to make it even easier find could just also return the name along side SinkBuilder basically the whole backend tuple.

Reply:
```rust
pub fn find(name: Option<String>) -> Option<&'static(&'static str, SinkBuilder)> {
    if let Some(name) = name {
        BACKENDS.iter().find(|backend| name == backend.0)
    } else {
        BACKENDS.first()
    }
}
```

Reply:
I probably would rather add a function than break the API though if we go down that road.

Reply:
1/2 done. I added ", or the current value when the Alsa mixer is used."

Reply:
> Same question as above as far as if it makes sense to expose alsa specific options even if it's compiled in if you're not actually using the alsa backend?

I see what you mean now. I do think it would be nice to bail out with an error. On the other hand, I think that in manu GNU tools you can specify all the parameters you want and they'll be silently ignored if they don't apply. Not to say that the GNU tools are the Golden Standard, but just as an example.

Reply:
I think we should exit if the backend is not alsa but alsa volume control is selected but just warn if alsa volume modifiers are called in combo with softvol.

Reply:
To clarify it's possible to use alsa hardware volume control in combination  with other backends. I'm not sure if that's a feature or a bug?

Reply:
Hmm, I think it's unintended but might be useful to some. After all, some other backend *may* in the end pipe to Also after all. Printing a warning seems a fine idea.

Reply:
Let's continue this in the thread below.